HGCal EGamma ID helpers and Phase-II MiniAOD 93x

nsmith- opened the pull request:
backport of #21037

nsmith- committed:
Add HGCal multicluster-derived EGamma to PAT

nsmith- committed:
Repackage HGCal EGamma identification tools

Originally https://github.com/CMS-HGCAL/EgammaTools
Moving into existing cmssw package to embed HGCal Egamma objects into
phase 2 miniAOD production.

nsmith- committed:
Add gsf track and pt 15 cut

nsmith- committed:
code checks

nsmith- committed:
Speed up HGCal EGamma isolation

cmsbuild commented:
A new Pull Request was created by  @nsmith- (Nicholas Smith) for CMSSW_9_3_X.

It involves the following packages:

PhysicsTools/PatAlgos
RecoEgamma/EgammaTools


@perrotta, @cmsbuild, @monttj, @slava77 can you please review it and eventually sign? Thanks.
@rappoccio, @gouskos, @jainshilpi, @TaiSakuma, @varuns23, @seemasharmafnal, @mmarionncern, @imarches, @ahinzmann, @acaudron, @lgray, @jdolen, @ferencek, @Sam-Harper, @jdamgov, @nhanvtran, @gkasieczka, @schoef, @JyothsnaKomaragiri, @mverzett, @gpetruc, @mariadalfonso, @pvmulder this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

kpedro88 commented:
assign upgrade

kpedro88 commented:
please test

cmsbuild commented:
New categories assigned: upgrade

@kpedro88 you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a3910a31bcddfe1585f8337c70631e3eb07d48d0
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24013/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4a685b34a1523082b5601de2e39c7f1ad54695ad
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24013/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24013/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24013/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2708283
 - DQMHistoTests: Total failures: 239
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2707857
 - DQMHistoTests: Total skipped: 187
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -27,6 +27,13 @@ def miniAOD_customizeCommon(process):
+    # so add companion collections for Phase-II MiniAOD production
+    from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
+    from PhysicsTools.PatAlgos.slimming.addHGCalEgammaCollections import addHGCalEgammaCollections
+    if phase2_hgcal.isChosen():
```
@kpedro88 
I recall that you mentioned (in ORP?) that 93X updates to miniAOD will come without modifying the default miniAOD of 93X.
This PR changes the default.
Please clarify.

kpedro88 replied:
@slava77 though the use of eras can be improved (as commented in the master branch PR), this doesn't change the default 93X miniAOD.

slava77 commented:
On 10/31/17 5:54 PM, Kevin Pedro wrote:
> *@kpedro88* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py
> <https://github.com/cms-sw/cmssw/pull/21045#discussion_r148020927>:
> 
>> @@ -27,6 +27,13 @@ def miniAOD_customizeCommon(process):
>  
>      process.patMuons.computeMiniIso = cms.bool(True)
>      
> +    # EGamma objects from HGCal are not yet in GED
> +    # so add companion collections for Phase-II MiniAOD production
> +    from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
> +    from PhysicsTools.PatAlgos.slimming.addHGCalEgammaCollections import addHGCalEgammaCollections
> +    if phase2_hgcal.isChosen():
> 
> @slava77 <https://github.com/slava77> though the use of eras can be
> improved (as commented in the master branch PR), this doesn't change the
> default 93X miniAOD.


are you saying that phase2_hgcal is not used in the standard upgrade
workflows for miniAOD?


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/21045#discussion_r148020927>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcblSj0EdhSlPiWCLrGYaQBuordRfKks5sxzSwgaJpZM4QJAL5>.
>

kpedro88 commented:
@slava77 I guess I'm confused about your comment. I thought by "default miniAOD" you meant Run2. Of course this will change the upgrade miniAOD for 93X, that is the point of the PR.

My comment in the ORP only covered MIP timing-related changes.

slava77 commented:
On 10/31/17 6:35 PM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> I guess I'm confused about your
> comment. I thought by "default miniAOD" you meant Run2. Of course this
> will change the upgrade miniAOD for 93X, that is the point of the PR.
> 
> My comment in the ORP only covered MIP timing-related changes.

"default" for me is everything in the current release covered by
runTheMatrix

OK, so the answer from you is that we can expect changes in the upgrade wfs
for 93X.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/21045#issuecomment-340801787>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbnsK-MGQbY2U6HxS9gpf0DtdEfbNks5sxz5IgaJpZM4QJAL5>.
>

nsmith- committed:
code review

nsmith- committed:
more code review

nsmith- committed:
Explain hgcal shower depth

nsmith- committed:
Make slimmedEgammaFromMultiClTask and import it into miniAOD_tools

Also allows someone to use slimmedEgammaFromMultiCl_cff separately

nsmith- committed:
Remove variables that are embedded in pat::Electron

Electron cluster is the one exception, as it needs some PAT code changes
to allow clusters where the seed is in ForwardSubdetector

nsmith- committed:
more code review

nsmith- committed:
Merge HGCal electron and photon ID helpers, more code review

nsmith- committed:
Add the duplicate hgcal electron filter

Also drop pt cut back to 10 because we don't have a billion fake
electrons now.

nsmith- committed:
code checks

nsmith- committed:
Functional fillDescriptions & code review

nsmith- committed:
Clone slimmedXFromMultiCl from corresponding slimmedX

In case new required config options are introduced.

kpedro88 commented:
@nsmith- please bring this up to date with #21037

cmsbuild commented:
Pull request #21045 was updated. @perrotta, @cmsbuild, @monttj, @kpedro88, @slava77 can you please check and sign again.

nsmith- committed:
Wrap hgcal-specific ID helpers with namespace, trim parameters

cmsbuild commented:
Pull request #21045 was updated. @perrotta, @cmsbuild, @monttj, @kpedro88, @slava77 can you please check and sign again.

nsmith- committed:
indentation

nsmith- committed:
Keep egamma from multiclusters in Phase-II AODSIM

Add a note about some duplicate event content location

slava77 commented:
@nsmith- 
please update this PR to by in sync with #21037

cmsbuild commented:
Pull request #21045 was updated. @perrotta, @cmsbuild, @monttj, @kpedro88, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 91767d3a35d3cd50451fe2ed6620a7abc5f2a0c1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24633/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/ac8c8135004230b02a771d59ee52e68ad72bf215
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24633/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24633/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24633/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2708283
 - DQMHistoTests: Total failures: 238
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2707858
 - DQMHistoTests: Total skipped: 187
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 8 edm output root files, 26 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/21045	91767d3
- backport for miniAOD production for HGCAL "FromMultiCl". Changes in the code are the same as in the master branch PR #21037 
- jenkins tests pass
- see more details in the signoff of #21037 https://github.com/cms-sw/cmssw/pull/21037#issuecomment-346484906

kpedro88 commented:
+1

nsmith- committed:
Further code review

cmsbuild commented:
Pull request #21045 was updated. @perrotta, @cmsbuild, @monttj, @kpedro88, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21045/24735/summary.html

There are some workflows for which there are errors in the baseline:
 136.788 step 3
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2572974
 - DQMHistoTests: Total failures: 232
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2572563
 - DQMHistoTests: Total skipped: 179
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 103 log files, 8 edm output root files, 25 DQM output files

kpedro88 commented:
+1

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/21045	1b7ef0f
- re-signing after a technical update since the last signoff by reco; changes are the same as in the master branch PR #21037 
- jenkins tests pass

davidlange6 commented:
merge
